lesson20/src/js/main.js

```diff
@@ -0,0 +1,106 @@
+import '../css/style.css'
+import * as loading from "./module/loading";
+import { createAttributedElements} from "./utils/createAttributedElements";
+
+const endPointURL = {
+    usersData: "https://myjson.dit.upm.es/api/bins/i55y"
+}
+
+const createErrorMessage = (error,element) => {
+    const errorMessage = createAttributedElements({
+        tag:"p",
+        valuesByAttributes:{
+            class:"error-message"
+        },
+        str:error
+    });
+    element.appendChild(errorMessage);
+}
+
+const fetchData = async (endpointURL,element) => {
+    const response = await fetch(endpointURL);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage("Communication with the server is broken.",element);
+        return;
+    }
+    const json = await response.json();
+    return json.data;
+}
+
+const fetchContentsData = (endPointURL,element,ms) => new Promise(resolve => setTimeout(() => resolve(fetchData(endPointURL,element)),ms));
+
+const table = document.getElementById("js-table");
+
+const initUsersData = async () => {
+    loading.showLoading(table);
+    try{
+        const usersData = await fetchContentsData(endPointURL.usersData,table,3000);
+        if (!usersData.length) {
+            createErrorMessage("No user.",table);
+            return;
+        }
+        sortUsersDataByIdAsc(usersData);
+        renderTableContents(usersData);
+    }catch(error){
+        console.error(error);
+    }finally{
+       loading.removeLoading(table);
+    }
+}
+
+const sortUsersDataByIdAsc = (usersData) =>  usersData.sort((a,b) => a.id - b.id); 
+
```

Comment:
IMO
> その時、idは要素の並び順でつけず以下のように順不同にしてください

I don't think sorting is necessary for this specification.Display in the same order as returned from the server (in no particular order).

Reply:
Thank you for your comment!
The IDs are in no particular order, 
but the images were sorted by ID, so I thought it needed to be sorted. Is this wrong?

![68747470733a2f2f74657272616365746563682e6a702f77702d636f6e74656e742f75706c6f6164732f323032312f30322fe38282e3828ae38191e38293e38197e38299e38285e3818f2d50312d322e706e67](https://user-images.githubusercontent.com/99260576/182262562-6546d67f-67f5-475b-81bf-c93095275ba8.png)

Reply:
I got the same point from Sae! (I put in the sorting too😂) 
The specification is to display the data as received.

Reply:
@443n0511 
Thank you for your comment!
I also misunderstood:scream:
Thank you for creating a PR about the specifications.

I'm going to fix it so that it doesn't sort!

Reply:
I fixed.
https://github.com/nakagawamai/js-lesson/pull/22/commits/118c748fe70323c3879c6cdb5ce7cca83d7afc28